### ISSUE TITLE
MongoDB dependancy

### ISSUE THREAD

**Issue by ludan-av** [superscriptjs/superscript#400]
We are big users of ChatScript. 
ChatScript is great but a nodejs based bot stack would be easier to scale.

ChatScript uses files to store conversations which (we think) is very convenient and easier (for us) to host than MongoDB.

Is the dependancy between SuperScript and MongoDB very strong? Would it be easy or super hard to use files instead of MongoDB?
Would it be a good or a bad idea independantly of the difficulty of doing it?

**Comment by Samurais** [comment]
MongoDB is for Production usage to persistence data, if you have many instances, the application would read or write simultaneously, the filesystem is not a good option.

MongoDB has great advantages for SuperScript.

Robomongo come up with a GUI Tool for Mongodb management.
https://robomongo.org/

**Comment by MaffooBristol** [comment]
I agree that a _hard dependency_ on a specific database is not perfect. Even though using mongodb is definitely better if you want to scale, which you mention as a reason for moving from CS to SS. In fact, one of the best ways to scale CS is to move from file-based to Mongo.

I'd imagine the best course of action would be to write a pull request with a load of tests and request for it to be merged; but I don't know if enough people would find this an issue for the maintainers to do the work themselves, if that makes sense?
